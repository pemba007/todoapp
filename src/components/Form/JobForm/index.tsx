import React from "react";
import { Form, Input, Button, Checkbox } from "antd";

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 16 },
};
const tailLayout = {
  wrapperCol: { offset: 2, span: 16 },
};

const JobForm: React.FC = () => {
  const onFinish = (values: any) => {
    console.log("Success:", values);
  };

  const onFinishFailed = (errorInfo: any) => {
    console.log("Failed:", errorInfo);
  };

  return (
    <Form
      layout='vertical'
      name='job-form'
      //   initialValues={{ remember: true }}
      onFinish={onFinish}
    >
      <Form.Item
        label='Title'
        name='title'
        rules={[{ required: true, message: "Please input your username!" }]}
      >
        <Input />
      </Form.Item>

      <Form.Item name='description' label='Description'>
        <Input.TextArea />
      </Form.Item>
      <Form.Item {...tailLayout}>
        <Button type='primary' htmlType='submit'>
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default JobForm;
