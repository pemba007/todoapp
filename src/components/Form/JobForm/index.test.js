import React from "react";

import { shallow, mount } from "enzyme";
import JobForm from "./index";
import { Button } from "antd";

describe("JobForm", () => {
  test("should render without problem", () => {
    shallow(<JobForm />);
  });

  test("should submit without error", () => {
    let formWrapper = mount(<JobForm />);
    formWrapper
      .find("input#job-form_title")
      .simulate("change", { target: { value: "test" } });
    formWrapper.find("form#job-form").simulate("submit");
  });

  test("should submit without error on empty", () => {
    let formWrapper = mount(<JobForm />);
    formWrapper.find("form#job-form").simulate("submit");
    expect(formWrapper).toMatchSnapshot();
  });
});
