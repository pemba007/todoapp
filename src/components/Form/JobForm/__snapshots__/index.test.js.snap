// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobForm should submit without error on empty 1`] = `ReactWrapper {}`;
